client/rest-high-level/src/main/java/org/elasticsearch/client/migration/DeprecationInfoResponse.java

```diff
@@ -0,0 +1,208 @@
+                fieldName = parser.currentName();
+            } else if (CLUSTER_SETTINGS.getPreferredName().equals(fieldName)) {
+                clusterSettings.addAll(parseDeprecationIssues(parser));
+            } else if (NODE_SETTINGS.getPreferredName().equals(fieldName)) {
+                nodeSettings.addAll(parseDeprecationIssues(parser));
+            } else if (INDEX_SETTINGS.getPreferredName().equals(fieldName)) {
+                // parse out the key/value pairs
+                while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                    String key = parser.currentName();
+                    List<DeprecationIssue> value = parseDeprecationIssues(parser);
+                    if (value.size() > 0) { // only add indices that contain deprecation issues
+                        indexSettings.put(key, value);
+                    }
+                }
+            }
+        }
+        return new DeprecationInfoResponse(clusterSettings, nodeSettings, indexSettings);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        DeprecationInfoResponse that = (DeprecationInfoResponse) o;
+        return Objects.equals(clusterSettingsIssues, that.clusterSettingsIssues) &&
+            Objects.equals(nodeSettingsIssues, that.nodeSettingsIssues) &&
+            Objects.equals(indexSettingsIssues, that.indexSettingsIssues);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(clusterSettingsIssues, nodeSettingsIssues, indexSettingsIssues);
+    }
+
+    @Override
+    public String toString() {
+        return clusterSettingsIssues.toString() + ":" + nodeSettingsIssues.toString() + ":" + indexSettingsIssues.toString();
+    }
+
+    /**
+     * Information about deprecated items
+     */
+    public static class DeprecationIssue {
+
+        private static final ParseField LEVEL = new ParseField("level");
+        private static final ParseField MESSAGE = new ParseField("message");
+        private static final ParseField URL = new ParseField("url");
+        private static final ParseField DETAILS = new ParseField("details");
+
+        static final ConstructingObjectParser<DeprecationIssue, Void> PARSER =
+            new ConstructingObjectParser<>("deprecation_issue", true,
+                a -> new DeprecationIssue(Level.fromString((String) a[0]), (String) a[1], (String) a[2], (String) a[3]));
+
+        static {
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), LEVEL);
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), MESSAGE);
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), URL);
+            PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), DETAILS);
+        }
+
+        public enum Level {
+            NONE,
+            INFO,
```

Comment:
Soon (likely tomorrow) I'll have a PR removing `NONE` and `INFO` from this list server-side - whichever merges last will have to remove them from this list too.

Reply:
++ pls link the PR when u have it

Reply:
Aforementioned PR: https://github.com/elastic/elasticsearch/pull/36326

Reply:
looks like i won ;)